var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -199,6 +199,8 @@ def file_system_options(self):
 
     def setup_environment(self, spack_env, run_env):
         spec = self.spec
+        spack_env.unset('F90')
+        spack_env.unset('F90FLAGS')
```

Comment:
Just to confirm, because I can never remember, does this break if the variables aren't set to begin with, or just it just skip them?

Reply:
I would also add a comment, just copy the warning message from `mvapich2`'s configure script.

Reply:
I'm not really sure if you're referring to spack or mvapich. If spack fails unsetting variables that aren't set, that seems like spack's problem. For mvapich, the configure was failing just because `F90` was set for me. I don't have an easy way to test GCC for what happens because all my machines use GCC > 5 generally and mvapich requires GCC < 6.

Reply:
What I'm saying is, if you run `spack install mvapich2` and `F90`/`F90FLAGS` aren't set, will Spack crash?

Reply:
So, on my laptop, I ripped out all the dependencies to just see the mvapich2 configure with `F90` unset before `spack install mvapich2` and Spack doesn't crash.